tests/test_rpm_verifier.py

```diff
@@ -356,6 +452,20 @@ def _mock_generate_output(with_failures: bool = False) -> MagicMock:
 
         return _mock_generate_output
 
+    @pytest.fixture()
+    def create_generate_results_mock(
+        self, monkeypatch: MonkeyPatch
+    ) -> Callable[[], MagicMock]:
+        """Create a generate_results mock"""
+
+        def _mock_generate_results() -> MagicMock:
+            """Monkey-patched generate_results"""
+            mock = create_autospec(generate_results, return_value={"results": "res"})
+            monkeypatch.setattr(rpm_verifier, generate_results.__name__, mock)
+            return mock
+
+        return _mock_generate_results
```

Comment:
What if we just returned the mock instead of the callable to get the mock?
```suggestion
    def generate_results_mock(
        self, monkeypatch: MonkeyPatch
    ) -> MagicMock:
        """Monkey-patched generate_results"""
            
        mock = create_autospec(generate_results, return_value={"results": "res"})
        monkeypatch.setattr(rpm_verifier, generate_results.__name__, mock)
        return mock
```

Reply:
In that other case, we returned a function that could take inputs that were used for generating mocks with different values. Here it's constant. I think we should remove the additional wrapper.